site/ko/beta/tutorials/generative/dcgan.ipynb

```diff
@@ -0,0 +1,989 @@
+    "colab": {},
+    "colab_type": "code",
+    "id": "3t5ibNo05jCB"
+   },
+   "outputs": [],
+   "source": [
+    "# `tf.function`이 어떻게 사용되어지는지 주목해 주세요.\n",
+    "# 이 주석은 함수가 \"컴파일\"되어 지도록 합니다.\n",
+    "@tf.function\n",
+    "def train_step(images):\n",
+    "    noise = tf.random.normal([BATCH_SIZE, noise_dim])\n",
+    "\n",
+    "    with tf.GradientTape() as gen_tape, tf.GradientTape() as disc_tape:\n",
+    "      generated_images = generator(noise, training=True)\n",
+    "\n",
+    "      real_output = discriminator(images, training=True)\n",
+    "      fake_output = discriminator(generated_images, training=True)\n",
+    "\n",
+    "      gen_loss = generator_loss(fake_output)\n",
+    "      disc_loss = discriminator_loss(real_output, fake_output)\n",
+    "\n",
+    "    gradients_of_generator = gen_tape.gradient(gen_loss, generator.trainable_variables)\n",
+    "    gradients_of_discriminator = disc_tape.gradient(disc_loss, discriminator.trainable_variables)\n",
+    "\n",
+    "    generator_optimizer.apply_gradients(zip(gradients_of_generator, generator.trainable_variables))\n",
+    "    discriminator_optimizer.apply_gradients(zip(gradients_of_discriminator, discriminator.trainable_variables))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "2M7LmLtGEMQJ"
+   },
+   "outputs": [],
+   "source": [
+    "def train(dataset, epochs):\n",
+    "  for epoch in range(epochs):\n",
+    "    start = time.time()\n",
+    "\n",
+    "    for image_batch in dataset:\n",
+    "      train_step(image_batch)\n",
+    "\n",
+    "    # GIF를 위한 이미지를 바로 생성합니다.\n",
+    "    display.clear_output(wait=True)\n",
+    "    generate_and_save_images(generator,\n",
+    "                             epoch + 1,\n",
+    "                             seed)\n",
+    "\n",
+    "    # 에포크 15개가 지날 때마다 모델을 저장합니다.\n",
+    "    if (epoch + 1) % 15 == 0:\n",
+    "      checkpoint.save(file_prefix = checkpoint_prefix)\n",
+    "    \n",
+    "    # print (' 에포크 {} 에서 걸린 시간은 {} 초 입니다'.format(epoch +1, time.time()-start))\n",
+    "    print ('Time for epoch {} is {} sec'.format(epoch + 1, time.time()-start))\n",
+    "\n",
+    "  # 마지막 에포크가 끝난 후 생성합니다.\n",
+    "  display.clear_output(wait=True)\n",
+    "  generate_and_save_images(generator,\n",
+    "                           epochs,\n",
+    "                           seed)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "2aFF7Hk3XdeW"
+   },
+   "source": [
+    "**이미지 생성 및 저장**\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "RmdVsmvhPxyy"
+   },
+   "outputs": [],
+   "source": [
+    "def generate_and_save_images(model, epoch, test_input):\n",
+    "  # `training`이 False로 맞춰진 것을 주목하세요.\n",
+    "  # 이렇게 하면 모든 층들이 추론 모드 (배치정규화)로 실행되어집니다. \n",
```

Comment:
Because BatchNorm's parameter is modified by input data, it must be training=False in inference mode.
I suggest '이렇게 하면 모든 층들이 추론 모드 (배치정규화)로 실행되어집니다' --> '이렇게 하면 **(배치정규화를 포함하여)** 모든 층들이 추론 모드로 **실행됩니다**'